### ISSUE TITLE
Storybook: dropdown should be accessible

### ISSUE THREAD

**Issue by marcysutton** [forem/forem#11724]
**Describe the bug**

In going through the components in Storybook, I found a dropdown that needs to be made accessible. It resembles the "hover only button" mentioned in #7139, which makes me think that possibly the implementation on the live site was updated but Storybook was not. This dropdown only shows content on hover of a link that looks like a button, and it's not keyboard accessible. It should be a `<button>` element and show content when activated with the keyboard.

I worked on a dropdown implementation in #11509 that could be referenced for this work, particularly for keyboard events and focus management. It would be great to have a repeatable component that's accessible, so all instances could be brought up to date.

**To Reproduce**

1. Go to https://storybook.forem.com/?path=/story/components-dropdowns--default
2. Observe the button that says "Hover to trigger dropdown"
3. Try focusing it with your keyboard, and you'll find the content isn't accessible without a mouse.

**Expected behavior**

The button should be focusable but only toggle content on Enter/Space or the Escape keys. It should have some ARIA attributes that indicate it's a toggle control, such as `aria-haspopup` and `aria-expanded`.

**Screenshots**

<img width="1366" alt="Storybook with hover dropdown open in Element Inspector" src="https://user-images.githubusercontent.com/1045233/100973468-746c1300-34ef-11eb-9390-6337ddbb4331.png">

**Desktop (please complete the following information):**

- OS, version: All
- Browser, version: All

**Smartphone (please complete the following information):**

- Device: All
- OS, version: All
- Browser, version: All

**Additional context**

This is part of a broader review of Storybook components, so issues will be filed separately.
